monai/apps/deepgrow/transforms.py

```diff
@@ -0,0 +1,623 @@
+
+        self.guidance = guidance
+        self.spatial_size = list(spatial_size)
+        self.spatial_size_key = spatial_size_key
+        self.margin = margin
+        self.meta_key_postfix = meta_key_postfix
+        self.start_coord_key = start_coord_key
+        self.end_coord_key = end_coord_key
+        self.original_shape_key = original_shape_key
+        self.cropped_shape_key = cropped_shape_key
+
+    def bounding_box(self, points, img_shape):
+        ndim = len(img_shape)
+        margin = ensure_tuple_rep(self.margin, ndim)
+        for m in margin:
+            if m < 0:
+                raise ValueError("margin value should not be negative number.")
+
+        box_start = [0] * ndim
+        box_end = [0] * ndim
+
+        for di in range(ndim):
+            dt = points[..., di]
+            min_d = max(min(dt - margin[di]), 0)
+            max_d = min(img_shape[di] - 1, max(dt + margin[di]))
+            box_start[di], box_end[di] = min_d, max_d
+        return box_start, box_end
+
+    def __call__(self, data):
+        guidance = data[self.guidance]
+        box_start = None
+        for key in self.keys:
+            box_start, box_end = self.bounding_box(np.array(guidance[0] + guidance[1]), data[key].shape[1:])
+            center = np.mean([box_start, box_end], axis=0).astype(int).tolist()
+            spatial_size = data.get(self.spatial_size_key, self.spatial_size)
+
+            current_size = np.absolute(np.subtract(box_start, box_end)).astype(int).tolist()
+            spatial_size = spatial_size[-len(current_size) :]
+            if len(spatial_size) < len(current_size):  # 3D spatial_size = [256,256] (include all slices in such case)
+                diff = len(current_size) - len(spatial_size)
+                spatial_size = list(data[key].shape[1 : (1 + diff)]) + spatial_size
+
+            if np.all(np.less(current_size, spatial_size)):
+                if len(center) == 3:
+                    center[0] = center[0] + (spatial_size[0] // 2 - center[0])
+                cropper = SpatialCrop(roi_center=center, roi_size=spatial_size)
+            else:
+                cropper = SpatialCrop(roi_start=box_start, roi_end=box_end)
+            box_start, box_end = cropper.roi_start, cropper.roi_end
+
+            meta_key = f"{key}_{self.meta_key_postfix}"
+            data[meta_key][self.start_coord_key] = box_start
+            data[meta_key][self.end_coord_key] = box_end
+            data[meta_key][self.original_shape_key] = data[key].shape
+
+            image = cropper(data[key])
+            data[meta_key][self.cropped_shape_key] = image.shape
+            data[key] = image
+
+        pos_clicks, neg_clicks = guidance[0], guidance[1]
+        pos = np.subtract(pos_clicks, box_start).tolist() if len(pos_clicks) else []
+        neg = np.subtract(neg_clicks, box_start).tolist() if len(neg_clicks) else []
+
+        data[self.guidance] = [pos, neg]
+        return data
+
+
+class ResizeGuidanced(Transform):
```

Comment:
should be a MapTransform?

Reply:
i don't see any guideline on using maptransform all the time... for bringing your own transforms (deepgrow app in this case)